src/firmware/linux/ioctl/guest.rs

```diff
@@ -0,0 +1,62 @@
+use std::marker::PhantomData;
+
+use crate::firmware::types::*;
+use crate::impl_const_id;
+
+use iocuddle::*;
+
+// These enum ordinal values are defined in the Linux kernel
+// source code: include/uapi/linux/sev-guest.h
+impl_const_id! {
+    pub Id => u32;
+    SnpGetReport<'_, '_> = 0,
+    SnpGetDerivedKey = 1,
+    SnpGetExtReport = 2,
+}
+
+const SEV: Group = Group::new(b'S');
+
+pub const SNP_GET_REPORT: Ioctl<WriteRead, &Command<SnpGetReport>> = unsafe { SEV.write_read(0) };
+pub const SNP_GET_DERIVED_KEY: Ioctl<WriteRead, &Command<SnpGetDerivedKey>> =
+    unsafe { SEV.write_read(0) };
+pub const SNP_GET_EXT_REPORT: Ioctl<WriteRead, &Command<SnpGetExtReport>> =
+    unsafe { SEV.write_read(0) };
```

Comment:
My attempts to use this failed. In the host side (ioctl to `/dev/sev`) there's only one ioctl (number 0), which accept one struct (`sev_issue_cmd`, here `Command`), which then has the ID of the actual command to run and a pointer to the data.

But in the guest side (ioctl to `/dev/sev-guest`) it's different: there are three different ioctls:

0 = SNP_GET_REPORT
1 = SNP_GET_DERIVED_KEY
2 = SNP_GET_EXT_REPORT

And each one of them accepts different structs. So I think that here (in `guest.rs`) it should be:

```rust
pub const SNP_GET_REPORT: Ioctl<WriteRead, &SnpGetReport> = 
    unsafe { SEV.write_read::<SnpGetReport>(0x0) };
pub const SNP_GET_DERIVED_KEY: Ioctl<WriteRead, &SnpGetDerivedKey> =
    unsafe { SEV.write_read::<SnpGetDerivedKey>(0x1) };
pub const SNP_GET_EXT_REPORT: Ioctl<WriteRead, &SnpGetExtReport> =
    unsafe { SEV.write_read::<SnpGetExtReport>(0x2) };
```

(maybe the numbers in the final parens can be replaced with `SnpGetReport::ID` etc.)

The `struct Command` should be removed from `guest.rs`.